lib/jpm/core.js

```diff
@@ -0,0 +1,235 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
All of these functions here need documentation -- I had no idea what they were doing until I looked at tests. They're all for the gcli hooks?